src/components/Calculator.vue

```diff
@@ -1,168 +1,320 @@
+                }
 
-      strong {
-        font-size: 2em;
-        letter-spacing: 2px;
-      }
-    }
+                if (this.operation) {
+                    this.second_argument += digit;
+                } else {
+                    this.first_argument += digit;
+                }
+                this.result();
+            },
 
-    .two-horizontal-cells {
-      grid-area: large_button;
-    }
+            onDotClick() {
+                let regExpForDot = /\./;
+
+                if (!this.operation) {
+                    this.first_argument += !this.first_argument ? '0' : '';
+
+                    this.first_argument += this.isValueExist(this.first_argument, regExpForDot) != -1 ? '' : '.';
+                } else {
+                    this.second_argument += !this.second_argument ? '0' : '';
+                    this.second_argument += this.isValueExist(this.second_argument, regExpForDot) != -1 ? '' : '.';
+                }
+
+                this.result();
+            },
+
+            onOperationClick(operation) {
+
+                if (!this.first_argument) {
+                    this.first_argument = '0';
+                }
+
+                this.first_argument = this.deleteDotIfNoDigitAfter(this.first_argument)
+
+                if (this.second_argument) {
+                    this.second_argument = this.deleteDotIfNoDigitAfter(this.second_argument)
+                    this.onGetResult();
+                }
+
+                this.operation = operation;
+                this.result();
+            },
+
+            deleteDotIfNoDigitAfter(argument) {
+
+                if (this.isValueExist(argument, /\./) == (argument.length - 1)) {
+                    return argument.slice(0, argument.length - 1)
+                } else {
+                    return argument
+                }
+            },
+
+            onGetResult() {
+
+                try {
+                    if (!this.first_argument || (!this.first_argument && !this.second_argument)) {
+                        return;
+                    }
+
+                    if (!this.operation && !this.lastOperation) {
+                        return;
+                    }
+
+                    if (this.first_argument && !this.operation && !this.second_argument && this.lastOperation) {
+                        this.operation = this.lastOperation;
+                        this.second_argument = this.lastSecond_argument;
+                    }
+
+                    if (this.first_argument && this.operation && !this.second_argument) {
+                        this.second_argument = this.first_argument;
+                    }
+
+                    if (this.second_argument === '0' && this.operation === '/') {
+                        throw new Error('Can not divide by zero');
+                    }
+
+                    if (!this.second_argument && this.operation) {
+                        this.second_argument += this.first_argument;
```

Comment:
can't catch in what case this logic will work, and what is expected result

Reply:
this condition duplicates the 160-line condition. delete it.